engine/launcher/general.cpp

```diff
@@ -85,174 +93,208 @@ char *split_words(char *string, int max_words) {
-	char *alloc;
-	alloc = (char *)malloc(strlen(string)+1);
-	if (alloc == 0) { ShowError("Out of memory", "G01", 1); return NULL; }
-	strncpy(alloc, string, strlen(string));
-	alloc[strlen(string)] = '\0';
-	return alloc;
+char *ptr_copy(char *string)
+{
+    char *alloc;
+    alloc = (char *) malloc(strlen(string) + 1);
+    if (alloc == 0) {
+        ShowError("Out of memory", "G01", 1);
+        return NULL;
+    }
+    strncpy(alloc, string, strlen(string));
+    alloc[strlen(string)] = '\0';
+    return alloc;
 }
 
 /* chomp
  * This function replaces trailing \n (carriage return) with the null character
  */
-void chomp(char *line) {
-	int current;
-	for (current = strlen(line) -1; (line[current] == '\n' || line[current] == ' ' || line[current] == 13); current--) {
-		line[current] = '\0';
-	}
+void chomp(char *line)
+{
+    int current;
+    for (current = strlen(line) - 1; (line[current] == '\n' || line[current] == ' ' || line[current] == 13);
+         current--) {
+        line[current] = '\0';
+    }
 }
+
 /* pre_chomp
  * This function Removes spaces and = from the front of the line and
  * returns a pointer to the new starting character
  */
 
-char *pre_chomp(char *line) {
-	while (line[0] == '=' || line[0] == ' ' || line[0] == 13 || line[0] == 9) { line++; }
-	return line;
+char *pre_chomp(char *line)
+{
+    while (line[0] == '=' || line[0] == ' ' || line[0] == 13 || line[0] == 9) {
+        line++;
+    }
+    return line;
 }
 
 /* replace
  * This function searches the first parameter for the second parameter
  * If the second parameter is found, it is replaced with the third parameter
  */
 
-char *replace(char *line, char *search, char *replace, int LENGTH) {
-	int length, dif, calc;
-	char * chr_new, * current;
-	char *ptr_new, *location;
-	calc = strlen(line) - strlen(search) + strlen(replace);
-	if (calc > LENGTH) { return line; }
-	chr_new= new char [LENGTH+1];
-	current= new char [LENGTH+1];
-	length = strlen(line);
-	strcpy(current, line);
-	while ((location = strstr(current, search)) > 0) {
-		chr_new[0] = '\0';
-		calc = strlen(current) - strlen(search) + strlen(replace);
-		if (calc > LENGTH) { strcpy(line, current); return line; }
-		dif = location - current;
-		strncat(chr_new, current, dif);
-		strncat(chr_new, replace, strlen(replace));
-		ptr_new = current + dif + strlen(search);
-		strncat(chr_new, ptr_new, strlen(ptr_new));
-		strcpy(current, chr_new);
-	}
-	strcpy(line, current);
-	delete [] chr_new;
-	delete [] current;
-	return line;
+char *replace(char *line, char *search, char *replace, int LENGTH)
+{
+    int length, dif, calc;
+    char *chr_new, *current;
+    char *ptr_new, *location;
+    calc = strlen(line) - strlen(search) + strlen(replace);
+    if (calc > LENGTH) {
+        return line;
+    }
+    chr_new = new char[LENGTH + 1];
+    current = new char[LENGTH + 1];
+    length = strlen(line);
+    strcpy(current, line);
+    while ((location = strstr(current, search)) > 0) {
```

Comment:
not saying we should fix this now; but we should certainly include these kinds of assignments in our linting to flag and fix.